src/main/java/rx/exceptions/Exceptions.java

```diff
@@ -148,5 +147,32 @@ public static Throwable getFinalCause(Throwable e) {
         }
         return e;
     }
-
+    /**
+     * Throws a single or multiple exceptions contained in the collection, wrapping it into
+     * {@code CompositeException} if necessary.
+     * @param exceptions the collection of exceptions. If null or empty, no exception is thrown.
+     * If the collection contains a single exception, that exception is either thrown as-is or wrapped into a
+     * CompositeException. Multiple exceptions are wrapped into a CompositeException.
+     * @param whileText the circumstance string to be appended to the thrown CompositeException, inserted after
+     * the sentences "Exception" and "Multiple exceptions".
+     */
+    public static void throwIfAny(Collection<? extends Throwable> exceptions, String whileText) {
+        if (exceptions != null && !exceptions.isEmpty()) {
+            if (exceptions.size() == 1) {
+                Throwable t = exceptions.iterator().next();
+                if (t instanceof RuntimeException) {
+                    throw (RuntimeException) t;
+                } else
+                if (t instanceof Error) {
+                    throw (Error) t;
+                } else {
+                    throw new CompositeException(
+                            "Exception" + whileText, exceptions);
+                }
+            } else {
+                throw new CompositeException(
+                        "Multiple exceptions" + whileText, exceptions);
+            }
```

Comment:
As per the discussion at https://github.com/ReactiveX/RxJava/issues/2602 ... this is an example where @akarnokd and I have slightly different styles, but it's okay. I always have `else if` on the same like, here it is split, but I can still read the code and it works so I choose to not make a big deal out of it. 

Sometimes when @akarnokd and I end up modifying each others code we'll see some lines get modified to personal style if we touch them, but we don't go around explicitly trying to change code just to match a personal preference ... if for no other reason than to not clutter changelogs and diffs.